services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/task.py

```diff
@@ -77,7 +79,7 @@ class DynamicSidecarsScheduler:  # pylint: disable=too-many-instance-attributes
 
     _lock: Lock = field(default_factory=Lock)
     _to_observe: dict[ServiceName, SchedulerData] = field(default_factory=dict)
-    _service_observation_task: dict[ServiceName, asyncio.Task] = field(
+    _service_observation_task: dict[ServiceName, Union[asyncio.Task, object]] = field(
```

Comment:
I think it is more clear to have ``None`` here provided that there are only two states:
```python
dict[ServiceName, Optional[asyncio.Task]]
```
and if you like you can keep ``_DISABLED_MARK=None``.

The approach you use would make more sense if ``None`` is already used to identify a state (i.e. when you need more than two states).

Reply:
> The approach you use would make more sense if `None` is already used to identify a state (i.e. when you need more than two states).

This is actually the case here, that's why I have to add a different marker. `None` is already used.

Reply:
if that is the case, then why ``Union[asyncio.Task, object]`` and not ``Optional[asyncio.Task, object]``?